site/app/controllers/admin/PlagiarismController.php

```diff
@@ -629,129 +679,262 @@ public function saveNewPlagiarismConfiguration(string $new_or_edit, string $grad
+            try {
+                $config_id = $em->getRepository(PlagiarismConfig::class)
+                    ->findOneBy(["gradeable_id" => $gradeable_id], ["config_id" => "DESC"]);
+                if ($config_id === null) {
+                    $config_id = 1;
+                }
+                else {
+                    $config_id = $config_id->getConfigID() + 1;
+                }
+            }
+            catch (Exception $e) {
+                $this->core->addErrorMessage($e->getMessage());
+                return new RedirectResponse($return_url);
+            }
+        }
 
-        // Version /////////////////////////////////////////////////////////////
-        $version_option = $_POST['version_option'] ?? "";
+        // Save configuration form data
+        if ($new_or_edit === "new" && isset($_POST["config_option"]) && $_POST["config_option"] === "upload_config") { // uploaded config file
+            // error checking
+            if (empty($_FILES) || !isset($_FILES["upload_config_file"]) || !isset($_FILES["upload_config_file"]["tmp_name"]) || $_FILES["upload_config_file"]["tmp_name"] === "") {
+                $this->core->addErrorMessage("Error: File upload failed");
+                return new RedirectResponse($return_url);
+            }
+            // load, parse, and save the config info
+            try {
+                $data = json_decode(file_get_contents($_FILES["upload_config_file"]["tmp_name"]), true);
 
+                // This is a little ugly/repetitive but it can be frustrating for users to get nondescriptive errors
+                // so we try to make potential error cases a little more helpful
+                $error_message = "";
+                if (!isset($data["version"])) {
+                    $error_message = "Error: Invalid or missing version field";
+                }
+                elseif (!isset($data["regex"])) {
+                    $error_message = "Error: Invalid or missing regex field";
+                }
+                elseif (!isset($data["regex_dirs"])) {
+                    $error_message = "Error: Invalid or missing regex_dirs field";
+                }
+                elseif (!isset($data["language"])) {
+                    $error_message = "Error: Invalid or missing language field";
+                }
+                elseif (!isset($data["threshold"])) {
+                    $error_message = "Error: Invalid or missing threshold field";
+                }
+                elseif (!isset($data["sequence_length"])) {
+                    $error_message = "Error: Invalid or missing sequence_length field";
+                }
+                elseif (!isset($data["prior_term_gradeables"])) {
+                    $error_message = "Error: Invalid or missing prior_term_gradeables field";
+                }
+                elseif (!isset($data["ignore_submissions"])) {
+                    $error_message = "Error: Invalid or missing ignore_submissions field";
+                }
```

Comment:
How about something like this to make this shorter:
```suggestion
                $error_message = "";
                $keys = ["version", "regex", "regex_dirs", "language", "threshold", "sequence_length", "prior_term_gradeables", "ignore_submissions"];
                for ($keys as $key) {
                    if (!isset($data[$key])) {
                        $error_message .= "Error: Invalid or missing {$key} field";
                    }
                }
```